### PR TITLE
[DOCS] Move CCR overview pages

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/elasticsearch/pull/34726 and https://github.com/elastic/stack-docs/pull/132

This PR removes the CCR overview pages from the Elasticsearch Reference (since they exist in the Stack Overview here: https://www.elastic.co/guide/en/elastic-stack-overview/master/xpack-ccr.html).  It also updates the title of the CCR section so that it aligns with the other titles.

### CODE REVIEW

File: docs/reference/index.asciidoc [hunk 1]
```diff
@@ -55,8 +55,6 @@ include::index-modules.asciidoc[]
 
 include::ingest.asciidoc[]
 
-include::ccr/index.asciidoc[]
```
Comment (jasontedor) [inline docs/reference/index.asciidoc]: :+1:

File: docs/reference/ccr/index.asciidoc [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [role="xpack"]
 [testenv="platinum"]
 [[xpack-ccr]]
-= Cross-cluster replication
+= Replicating data across multiple clusters
```
Comment (jasontedor) [inline docs/reference/ccr/index.asciidoc]: I'm curious what is the reason for this change?

**Review by jasontedor — COMMENTED** [review]
I left a question.

File: docs/reference/ccr/index.asciidoc (hunk 2 above)
Comment (lcawl) [reply docs/reference/ccr/index.asciidoc]: It's not a necessity, but it seemed to align better with the titles of similar sections in the table of contents ("securing", "alerting", "monitoring").

Comment (jasontedor) [reply docs/reference/ccr/index.asciidoc]: I think I prefer "Cross-cluster replication". It's easier to discover (`grep`, searches, etc.).

Comment (lcawl) [reply docs/reference/ccr/index.asciidoc]: Ok!  I've reverted that change.

**Review by jasontedor — APPROVED** [review]
LGTM.
